t/t1017-update-db.t

```diff
@@ -96,70 +87,15 @@ test_expect_success 'get all the columns from the queue_table and make sure the
 	test_cmp queue_table_columns.expected queue_table_columns.test
 '
 
-test_expect_success 'successfully update flux-accounting DB from v0.10.0' '
-	cp ${DB_v0_10_0} tmp_v0_10_0.db &&
-	chmod 666 tmp_v0_10_0.db &&
-	flux account-update-db -p tmp_v0_10_0.db
-'
-
-test_expect_success 'successfully update flux-accounting DB from v0.11.0' '
-	cp ${DB_v0_11_0} tmp_v0_11_0.db &&
-	chmod 666 tmp_v0_11_0.db &&
-	flux account-update-db -p tmp_v0_11_0.db
-'
-
-test_expect_success 'successfully update flux-accounting DB from v0.12.0' '
-	cp ${DB_v0_12_0} tmp_v0_12_0.db &&
-	chmod 666 tmp_v0_12_0.db &&
-	flux account-update-db -p tmp_v0_12_0.db
-'
-
-test_expect_success 'successfully update flux-accounting DB from v0.13.0' '
-	cp ${DB_v0_13_0} tmp_v0_13_0.db &&
-	chmod 666 tmp_v0_13_0.db &&
-	flux account-update-db -p tmp_v0_13_0.db
-'
-
-test_expect_success 'successfully update flux-accounting DB from v0.14.0' '
-	cp ${DB_v0_14_0} tmp_v0_14_0.db &&
-	chmod 666 tmp_v0_14_0.db &&
-	flux account-update-db -p tmp_v0_14_0.db
-'
-
-test_expect_success 'successfully update flux-accounting DB from v0.15.0' '
-	cp ${DB_v0_15_0} tmp_v0_15_0.db &&
-	chmod 666 tmp_v0_15_0.db &&
-	flux account-update-db -p tmp_v0_15_0.db
-'
-
-test_expect_success 'successfully update flux-accounting DB from v0.16.0' '
-	cp ${DB_v0_16_0} tmp_v0_16_0.db &&
-	chmod 666 tmp_v0_16_0.db &&
-	flux account-update-db -p tmp_v0_16_0.db
-'
-
-test_expect_success 'successfully update flux-accounting DB from v0.17.0' '
-	cp ${DB_v0_17_0} tmp_v0_17_0.db &&
-	chmod 666 tmp_v0_17_0.db &&
-	flux account-update-db -p tmp_v0_17_0.db
-'
-
-test_expect_success 'successfully update flux-accounting DB from v0.18.0' '
-	cp ${DB_v0_18_0} tmp_v0_18_0.db &&
-	chmod 666 tmp_v0_18_0.db &&
-	flux account-update-db -p tmp_v0_18_0.db
-'
-
-test_expect_success 'remove temporary test DBs' '
-	rm tmp_v0_10_0.db &&
-	rm tmp_v0_11_0.db &&
-	rm tmp_v0_12_0.db &&
-	rm tmp_v0_13_0.db &&
-	rm tmp_v0_14_0.db &&
-	rm tmp_v0_15_0.db &&
-	rm tmp_v0_16_0.db &&
-	rm tmp_v0_17_0.db &&
-	rm tmp_v0_18_0.db
-'
+# test updating flux-accounting databases from older versions,
+# starting with v0.10.0
+for db in ${SHARNESS_TEST_SRCDIR}/expected/test_dbs/*; do
+	if [[ $db == *FluxAccounting* ]]; then
+		cp $db $(basename $db | cut -d '.' -f 1)_tmp.db
+		tmp_db=$(basename $db | cut -d '.' -f 1)_tmp.db
+		chmod +rw $tmp_db
+		test_expect_success 'update old DB: '$(basename $db) "flux account-update-db -p $tmp_db"
```

Comment:
One thought would be to test the updated db for validity after the update? (This wasn't in the original test so it is optional)

Reply:
Yeah, I think that would be a great next step. I wonder how I might be able to do this - maybe try to add some users and banks and make sure the operations are successful?

Reply:
I just squashed and pushed up a fix that runs both `add-bank` and `add-user` commands after updating each DB to make sure the command runs successfully. Let me know if that was along the lines of what you were thinking - I know you already approved but I just want to make sure!

Reply:
OK, after some digging, I also found some basic [integrity/validity checks](https://www.sqlite.org/pragma.html#pragma_integrity_check) that SQLite performs on the DB using a pragma. It looks for:

- Table or index entries that are out of sequence
- Misformatted records
- Missing pages
- Missing or surplus index entries
- `UNIQUE`, `CHECK`, and `NOT NULL` constraint errors

Since the test DB's only have a couple of entries in them, the check performs quickly and the result is now checked in the for-loop to make sure that the check is `"ok"`. I've squashed and pushed that up as well.

Reply:
That's cool! Nice work.

Reply:
Feel free to set MWP when you are ready, already approved :+1:

Reply:
Thanks! Will do